packages/files-ui/src/UI-components/AnchorMenu.tsx

```diff
@@ -0,0 +1,120 @@
+import React, {  ReactNode, useEffect, useMemo, useRef } from "react"
+import { MenuItem, Paper, PopoverPosition } from "@material-ui/core"
+import { makeStyles, createStyles, useOnClickOutside } from "@chainsafe/common-theme"
+import clsx from "clsx"
+import { CSFTheme } from "../Themes/types"
+
+interface Option {
+  contents: ReactNode
+  inset?: boolean
+  onClick?: (e: React.MouseEvent) => void
+  disabled?: boolean
+}
+
+interface Props {
+  isOpen: boolean
+  options: Option[]
+  anchorPosition?: PopoverPosition
+  onClose: () => void
+}
+
+type Position = {
+  top?: number
+  left?: number
+  bottom?: number
+  right?: number
+}
+
+
+const useStyles = makeStyles(({ constants, zIndex, animation }: CSFTheme) => {
+  return createStyles({
+    anchorRoot: (styleProps: Position) => ({
+      position: "fixed",
+      top: styleProps?.top,
+      left: styleProps?.left,
+      bottom: styleProps?.bottom,
+      right: styleProps?.right,
+      zIndex: zIndex?.blocker,
+      visibility: "hidden",
+      opacity: 0,
+      transition: `opacity ${animation.transform * 2}ms`,
+      "&.open": {
+        visibility: "visible",
+        opacity: 1
+      }
+    }),
+    paper: {
+      padding: `${constants.generalUnit}px 0`,
+      backgroundColor: `${constants.menu.backgroundColor} !important`,
+      color: `${constants.menu.color} !important`
+    },
+    options: {
+      "&:hover": {
+        backgroundColor: `${constants.menu.backgroundOptionHover} !important`
+      }
+    }
+  })})
+
+const MENU_RIGHT_SPACE_TOLERANCE = 180
+const MENU_BOTTOM_SPACE_TOLERANCE = 250
+
+export default function MenuDropdown({ isOpen, options, anchorPosition, onClose }: Props) {
+  const position = useMemo(() => {
+    const windowHeight = window.innerHeight
+    const windowWidth = window.innerWidth
+    const position: Position = {}
+    if (!anchorPosition) return {}
```

Comment:
An empty object is truthy, so better return undefined, and we can return earlier, and only if the menu is open.

```suggestion
    if (!anchorPosition || !isOpen) return

    const windowHeight = window.innerHeight
    const windowWidth = window.innerWidth
    const position: Position = {}
```